src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -0,0 +1,253 @@
+// <copyright file="Security.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Datadog.Trace.Agent;
+using Datadog.Trace.AppSec.Agent;
+using Datadog.Trace.AppSec.EventModel;
+using Datadog.Trace.AppSec.Transport;
+using Datadog.Trace.AppSec.Waf;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.AppSec
+{
+    /// <summary>
+    /// The Secure is responsible cooridating app sec
+    /// </summary>
+    public class Security : IDatadogSecurity
```

Comment:
Should this implement `IDisposable` since it owns a reference to `IPowerWaf`?

Reply:
fixed in 56eff08dc8899e65b62d0b8786e3b281851090b4